index.js

```diff
@@ -314,17 +327,27 @@ io.on('connection', function(socket){
                 if (userData.energy > 0) {
                     let cell = global.world.find(d => d.x === x && d.y === y);
                     if(cell && cell.owner === userData.security && cell.build !== builds.HQ){
-                        cell.owner = null;
-                        db.world.cellUpdate(x, y, null, cell.build, cell.level);
-                        io.emit('cell', x, y, null, null, cell.build, cell.level);
-
-                        userData.energy -= 1;
-                        userData.cells = utils.countPlayerCells(userData.security);
+                        if (cell.build !== builds.ROCK){
+                            cell.owner = null;
+                            db.world.cellUpdate(x, y, null, cell.build, cell.level);
+                            io.emit('cell', x, y, null, null, cell.build, cell.level);
 
-                        socket.emit('info', { energy: userData.energy, cells: userData.cells });
+                            userData.energy -= 1;
+                            userData.cells = utils.countPlayerCells(userData.security);
 
+                            socket.emit('info', { energy: userData.energy, cells: userData.cells });
+                        }
                         if(cell.build === builds.WAREHOUSE){
                             utils.updatePlayerMaxResources(userData);
+                        }else if(cell.build === builds.ROCK){
+                            cell.owner = null;
+                            db.world.cellUpdate(x, y, null, cell.build);
+                            io.emit('cell', x, y, null, null, cell.build);
+
+                            userData.energy -= 10;
+                            userData.cells = utils.countPlayerCells(userData.security);
+
+                            socket.emit('info', { energy: userData.energy, cells: userData.cells });
```

Comment:
Především tuhle část, ale i zbytek unclaim funkce, jsem vůbec nepochopil. Proč to zbytečně děláš složitější a na co odečítáš 10 energie za unclaim?

Reply:
Teraz som si to uvedomil, bude upravené.